.github/workflows/docs.yml

```diff
@@ -61,29 +61,46 @@ jobs:
           ref: gh-pages
           path: gh-pages
 
+      - name: Cleanup unrelated files
+        run: |
+          # Under `apps` we want to keep only the `README.md` files.  All other files
+          # (reframe tests, input files, etc...) should be ignored by the docs.
+          find benchmarks/apps -type f \! \( -name 'README.md' \) -print -delete
+
       - name: Build docs
         run: |
           if [[ ${{ github.event_name }} == 'pull_request' ]]; then
-            export MKDOCS_SITE_DIR="site/preview/PR${{ github.event.number }}"
-          else
-            export MKDOCS_SITE_DIR="site"
+              PREVIEW_SUBDIR="preview/PR${{ github.event.number }}"
+              echo "PREVIEW_SUBDIR=${PREVIEW_SUBDIR}" >> "${GITHUB_ENV}"
+              export MKDOCS_SITE_DIR="site/${PREVIEW_SUBDIR}"
+              export MKDOCS_SITE_URL="https://ukri-excalibur.github.io/excalibur-tests/${PREVIEW_SUBDIR}"
           fi
           mkdocs build
 
       - name: Deploy docs
         # Run only if push is to `main`, or if it's a PR not from a fork.
         if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') || (github.event_name == 'pull_request' && ! github.event.pull_request.head.repo.fork) }}
+        working-directory: gh-pages
         run: |
-          cd gh-pages
-          cp -fr ../site/* .
           git config user.name ${{github.actor}}
           git config user.email "${{github.actor_id}}+${{github.actor}}@users.noreply.github.com"
-          git add .
+          # Before copying the new files, delete the old ones, so that we can
+          # cleanly update the website.
           if [[ ${{ github.event_name }} == 'pull_request' ]]; then
-              COMMIT_MESSAGE="Update docs preview from ${{ github.sha }} for PR #${{ github.event.number }}"
+              COMMIT_MESSAGE="Update docs preview from ${{ github.sha }}"
+              # Only delete files in the preview subdir, if any
+              git rm -rf "${PREVIEW_SUBDIR}"
           else
               COMMIT_MESSAGE="Update docs from ${{ github.sha }}"
+              for file in $(git ls-files); do
+                  if [[ x"${file}" != xpreview/* ]]; then
```

Comment:
I don't understand this bit. You want to delete all files in the repository, except ones starting with `preview/`? Is the `x` just enforcing that `preview` is at the start of the path?

Reply:
Yes, and yes. The leading "x" is a common shell trick to ensure the comparison is done correctly, autotools scripts use it extensively for example.